kafka/src/kafka/impl/consumer_impl.cpp

```diff
@@ -292,6 +294,53 @@ void ConsumerImpl::Commit() { rd_kafka_commit(consumer_.GetHandle(), nullptr, /*
 
 void ConsumerImpl::AsyncCommit() { rd_kafka_commit(consumer_.GetHandle(), nullptr, /*async=*/1); }
 
+std::pair<std::int64_t, std::int64_t> ConsumerImpl::GetMinMaxOffset(const std::string& topic, std::int32_t partition) {
+    std::int64_t offset_high{-1};
+    std::int64_t offset_low{-1};
+
+    auto err = rd_kafka_query_watermark_offsets(
+        consumer_.GetHandle(), topic.data(), partition, &offset_low, &offset_high, /*timeout_ms=*/-1
+    );
+    if (err != RD_KAFKA_RESP_ERR_NO_ERROR) {
+        LOG_ERROR() << fmt::format("Failed to get offsets: {}", rd_kafka_err2str(err));
+        return {-1, -1};
+    }
+
+    if (offset_high == RD_KAFKA_OFFSET_INVALID) {
+        LOG_ERROR() << fmt::format("Invalid offset for topic {} partition {}", topic, partition);
+        return {-1, -1};
+    }
+
+    return {offset_low, offset_high};
+}
+
+std::vector<std::int32_t> ConsumerImpl::GetPartitionsId(const std::string& topic) {
+    const rd_kafka_metadata_t* raw_metadata{nullptr};
+    auto err = rd_kafka_metadata(consumer_.GetHandle(), 0, nullptr, &raw_metadata, /*timeout_ms=*/-1);
+    if (err != RD_KAFKA_RESP_ERR_NO_ERROR) {
+        LOG_ERROR() << fmt::format("Failed to fetch metadata: ", rd_kafka_err2str(err));
+        return {};
+    }
+    MetadataHolder metadata{raw_metadata};
```

Comment:
enclose the raw metadata logic in lambda function, like
```
MetadataHolder matadata{[&topic] {
    ...logic...
    return raw_metadata;
}()};
```

Reply:
Done